wiki/misc/codecs/en.md

```diff
@@ -0,0 +1,171 @@
+`error` returns a `PartialResult`, which allows you to both recover a decode, and to get the error message for why the decode failed. Right now, the `result` method is more important to us.
+`result` returns an `Option<Pair<A, T>>`, which makes sure that we know for sure if we have a result, otherwise we could just get null.
+
+Finally, we get to the last two lines:
+```java
+...
+.get() // Pair<Boolean, JsonElement>
+.getFirst(); // Boolean
+```
+
+We use `get` to unbox the `Option`. Generally this is unsafe to do, an IntelliJ even gives a warning.
+In this case we know that it is safe due to the simplicity of the example.
+Then finally, we call `getFirst` on `com.mojang.datafixers.util.Pair` to get the first half of the pair
+
+Wow. That sure was a lot.
+Now, I know this may seem like the `Codec` system is complicated right now, but unfortunately we have only scratched the surface.
+Yep. That's right, it gets so much worse.
+Let's step back and look at some more `Codec` types.
+</div>
+
+## Collection Codecs
+
+While the primative `Codec`s are the most basic building blocks for `Codec`s, we need to we able to put them together to be able to fully represent serializable objects.
+These collection `Codec`s are fairly straight forward, and each has a constructor which takes a `Codec` parameter for each associated type with the collection.
+
+<!-- TODO: Use the static methods instead of the classes -->
+
+- `ListCodec<T>`: A codec for `List<T>`. You can also make a list by calling `listOf` on a `Codec`.
+- `SimpleMapCodec<K, V>`: A codec for `Map<K, V>` with a known set of `K`s. This known set is an additional parameter.
+- `UnboundedMapCodec<K, V>`: A codec for `Map<K, V>`.
+- `PairCodec<F, S>`: A codec of a `Pair<F, S>`.
+- `EitherCodec<L, R>`: A codec of `Either<L, R>`.
+
+## The `RecordCodecBuilder`
+Oh no. There is an explicit type name in a header. This is going to get crazy.
+
+Let's start it off simple: a `RecordCodecBuilder` creates a `Codec` that can directly serialize and deserialize a Java object.
+While it has the name `Record` in it, this isnt specific to the `record`s in Java, but it's often a good idea to use `record`s.
+Going over a basic example will probably be the clearest here.
+
+```java
+record Foo(int bar, List<Boolean> baz, String qux) {
+    public static final Codec<Foo> CODEC =
+        RecordCodecBuilder.create(
+            instance ->
+                instance.group(
+                    Codec.INT.fieldOf("bar").forGetter(Foo::bar),
+                    Codec.BOOL.listOf().fieldOf("baz").forGetter(Foo::baz),
+                    Codec.STRING.optionalFieldOf("qux", "default string").forGetter(Foo::qux)
+                ).apply(instance, Foo::new)
+        );
+}
+```
+
+Ok, thats not too bad. 
+One nice thing about this is that Mojang did a lot of magic behind the scene to make this feel nice.
+Trust me, I (OroArmor) once wrote a similar library and partially gave up on doing the right thing.
```

Comment:
For the same reasons as above, I think you could rewrite this as "Some of us (like OroArmor) attempted to write equivalents, only to partially give up on doing the right things"

But also are these two sentences really necessary ? They feel a bit out of place in a wiki article.